src/main/java/org/folio/circulation/support/Clients.java

```diff
@@ -299,6 +310,18 @@ public CollectionResourceClient noteTypesClient() {
     return noteTypesClient;
   }
 
+  protected HttpClient getHttpClient() {
+    return httpClient;
+  }
+
+  protected RoutingContext getRoutingContext() {
+    return routingContext;
+  }
+
+  public CompletableFuture<Result<Drools>> getCirculationDrools() {
+    return CirculationRulesProcessor.getInstance().getDrools(routingContext, httpClient);
+  }
```

Comment:
Could these be moved to a separate class specific to circulation rules? That would mean that the clients class does not need to change dependencies and grow further.

Reply:
So, I added these methods at the end of the work because I was running into a situation where something like a hundred files would have to be changed and implementation details were spilling out across the code base.

This approach moves the routing context and http client one level deeper into Clients, but avoids moving them all the way into every class that uses Clients instead.

If there's a cleaner opportunity I missed. I'm open to it.

Reply:
@jsavell I've made some on an experimental branch from your code that reinstates the original dependencies for the Clients class and issued a [pull request](#704)

I'd appreciate your thoughts.

Reply:
Those changes clean things up. Please merge them in or make that the primary PR.